### ISSUE TITLE
2.0 updates for Bobassembly

### ISSUE THREAD

**Issue by Qatavin** [modded-factorio/bobsmods#200]
Quick notes

- [x] Fluid_box fix for new mechanics
- [x] --Add volume, remove old size parameters
- [x] --Add direction to pipe connections
- [x] --Adjust positions to meet new requirement to have connection withing bounding box
- [x] --Replace fluid_boxes.off_when_no_fluid_recipe with fluid_boxes_off_when_no_fluid_recipe
- [x] Emissions_per_minute fix for new pollution system
- [x] Fix working visualisations for chem plants
- [x] Switch graphics for hr_graphics
- [x] --Update assembling machine mask to higher resolution
- [x] See also #166
- [x] See also #167
- [x] Add wire connections to assemblers
- [x] Remove old low resolution graphics that are no longer used
- [x] Update all sounds to new system
- [x] Add impact_category

While I'm working on this, one thing I've been wanting to do for a while is add fluid boxes to electronics machine 1. It makes sense to do so because its recipe is closer to assembling machine 2 than 1. Also, it's kind of a pain that electronics machine 1 can't take ferric chloride solution, so you can't use it to make red circuit boards (which are required to make electronics machine 2)

**Comment by KiwiHawk** [comment]
I'm not sure about the electronics assembly machine 1 change. When talking about the recipe, are you looking before or after my recent changes to assembly machine tiers?

Yes, I know it's annoying how it is. I kinda like it 😈 Makes people appreciate the upgrade more!

**Comment by KiwiHawk** [comment]
Electronic assembly machines look good! What's the issue with distilleries? 🤔

Sure, if you want to update graphics. That's one thing I don't touch!

**Comment by Qatavin** [comment]
There's nowhere for the mirrored fluid box connection to attach to.
![distillery](https://github.com/user-attachments/assets/daada278-a6e4-4129-97f8-115eeadc0abe)

I'll take care of #167 at the same time as updating the mask, no problem.

I was trying to load Ores earlier (because I need Ores to load Plates, and Plates to have distilleries, electrolysers, and so on for what I was doing above)

**Comment by KiwiHawk** [comment]
I assumed that the distillery would just work as-is. It already has a line of symmetry. Flipping just means swapping the in and out.

**Comment by Qatavin** [comment]
That might work in some cases, but not all of them. And I think if the mirroring doesn't work in every case, the game won't allow it at all.

![distillery mirroring](https://github.com/user-attachments/assets/889e2e0f-2f41-4437-b06b-531c47f890e0)

Perhaps if the north version could be swapped with the south version when they're flipped, and east with west? Or something like that.

**Comment by Qatavin** [comment]
Turned out I was half right. It took a little research, but as it happens, the Recycler has a very similar issue with its item output, so the game has a graphics_set_flipped property to deal with this kind of situation. Distilleries have been solved.

![distillery mirroring win](https://github.com/user-attachments/assets/085c2e61-527f-4dbb-b241-6a16a56889fa)

**Comment by Qatavin** [comment]
Stone chemical furnaces turned out to be tougher than expected. This is what I ended up doing. It's definitely not perfect - you can see an incomplete section of pipe on the one in the top left. Could probably be fixed with a bit of a redraw, though. Or the graphics could be outright replaced. They _are_ very old.

![chem-furnace-flipped](https://github.com/user-attachments/assets/aa9bed59-58f0-460f-97e7-d951be656588)

Wire connections were annoying to figure out, but I think I've got the hang of them now. I'm not totally satisfied with this, but it gets the job done, I guess. If anyone has any thoughts about how to improve the positioning, I'm all ears.

![elecflip2](https://github.com/user-attachments/assets/3cc54c98-fb49-4045-9106-621805adf8e7)

Both of these feel like they'd look better if the back part of the pipe connection were smaller. But I feel like that's something that can be put off for later.

**Comment by Qatavin** [comment]
Once I got Plates running, there were a few more things I decided I needed to update, including air compressors and barreling pumps. Unfortunately...

![fluid-pump-connectors](https://github.com/user-attachments/assets/77070307-1c59-4bc1-a23b-58f411d37dcf)

By all appearances, there is no way whatsoever to make mirrored circuit connections. Which means that this is the best I can do, just having the connectors touching the entity in both mirrored and regular versions. It sucks, but I don't want to remove it because being able to control these with the circuit network seems like it might be pretty desirable, given their barreling and unbarreling applications. Any thoughts?
